### PR TITLE
verification: Add verification endpoints

### PR DESCRIPTION
**Pull request by flyingpimonster**
These endpoints provide information about how an app may be verified (based on its appid) and whether/how an app has been verified. Currently, apps may be verified (or blocked from being verified) by adding them to data/verification/{verified|blocked}.json or via website verification.

Part of #102.

---

See [this gist](https://gist.github.com/flyingpimonster/54be90a19e912e6b3e480da67a4d8103) for a description of the API endpoints. For testing website verification, you can use `net.jwestman.TestApp`.

### CODE REVIEW

File: app/verification.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            "detail": "malformed_app_id",
+        }
+
+    elif db.redis_conn.sismember("verification:blocked", appid):
+        return {
+            "methods": [],
+            "detail": "blocked_by_admins",
+        }
+
+    elif segments[0:2] in [["com", "github"], ["io", "github"]]:
```
Comment (razzeee) [inline app/verification.py:50]: I might be confused, but not sure (without trying) why this is `segments[0:2]` and not `segments[0:1]`

Comment (flyingpimonster) [reply app/verification.py:50]: The end index is exclusive.

File: app/utils.py [hunk 2]
```diff
@@ -202,3 +203,13 @@ def appstream2dict(reponame: str):
...
     return apps
+
+
+def get_appids(path):
+    try:
+        with open(
+            path,
+        ) as file_:
+            return json.load(file_)
+    except IOError:
```
Comment (danielsilverstone-ct) [inline app/utils.py:214]: Is there any value in logging the IOError here?  In case things stop working.

File: app/verification.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        with requests.Session() as session:
+            for verdict in ["verified", "blocked"]:
+                r = session.get(
+                    f"https://raw.githubusercontent.com/flathub/backend/master/data/verification/{verdict}.json"
+                )
+                if r.status_code == 200:
+                    value = r.json()
+                    p.unlink(f"verification:{verdict}")
+                    if len(value):
+                        p.sadd(f"verification:{verdict}", *r.json())
```
Comment (danielsilverstone-ct) [inline app/verification.py:19]:
```suggestion
                        p.sadd(f"verification:{verdict}", *value)
```

File: app/utils.py (hunk 2 above)
Comment (flyingpimonster) [reply app/utils.py:214]: Possibly? I didn't write this, just moved it from picks.py.

**Review by danielsilverstone-ct — APPROVED** [review]
I think this looks good, confirming it meets the particular requirements probably needs some one else, but from a code perspective I can't spot anything else to ask for changing.

**Pull request merged**